sdk/go/common/resource/plugin/host.go

```diff
@@ -376,12 +377,29 @@ func (host *defaultHost) ListAnalyzers() []Analyzer {
 }
 
 func (host *defaultHost) Provider(descriptor workspace.PackageDescriptor) (Provider, error) {
-	plugin, err := host.loadPlugin(host.loadRequests, func() (interface{}, error) {
+	load := func() (interface{}, error) {
 		pkg := descriptor.Name
 		version := descriptor.Version
 
-		// Try to load and bind to a plugin.
-
+		// There are cases where a provider may want to make use of configuration defined at a stack level, as well
```

Comment:
Odd diff? Isn't this on master already?

Reply:
Unfortunately not. An extended version was introduced in #17229; I figured I'd be cheeky and work a subset of that which describes the current state of affairs (no special treatment for dynamic providers) into this clean-up.